src/Core/Services/IOrganizationService.cs

```diff
@@ -31,8 +31,12 @@ public interface IOrganizationService
     Task DisableTwoFactorProviderAsync(Organization organization, TwoFactorProviderType type);
     Task<List<OrganizationUser>> InviteUsersAsync(Guid organizationId, Guid? invitingUserId,
         IEnumerable<(OrganizationUserInvite invite, string externalId)> invites);
+    Task<List<OrganizationUser>> InviteUsersAsync(Guid organizationId, Guid? invitingUserId,
+        IEnumerable<(OrganizationUserInvite invite, string externalId)> invites, EventSystemUser systemUser);
```

Comment:
I think the `systemUser` and `invitingUser` are mutually exclusive - in effect, the systemUser IS the "user" sending the invite. Please see if you can simplify this interface by removing `invitingUser`, and try to remove `invitingUser` from the shared private method as well.

Same with the other new interfaces in this service.

Reply:
Good point, I've implemented this.